src/mysql_conn.erl

```diff
@@ -158,8 +158,14 @@ connect(#state{connect_timeout = ConnectTimeout} = State) ->
     end.
 
 connect_socket(#state{tcp_opts = TcpOpts, host = Host, port = Port} = State) ->
+    %% 'nodelay' option is not supported for domain = local, but if inet_backend =/= socket this error will be ignored.
+    CanSetNodelay = case Host of
+        {local, _LocalAddr} -> false;
+        _NonLocalAddr -> true
+    end,
```

Comment:
Instead of a boolean, we can use meaningful values, for example:

```suggestion
    Transport = case Host of
        {local, _LocalAddr} -> local;
        _NonLocalAddr -> tcp
    end,
```

Reply:
Indeed，this modifications have more readability